_posts/2023-06-28-Remilia-interview.md

```diff
@@ -0,0 +1,68 @@
+
+I'm still missing about a dozen or so chips, some of which share a common emulation core. I expect these to take a bit longer to port because of how much more complex they are. Putting them off until later has also been beneficial because it means I now have more experience with porting emulation cores under my belt. But for now, I estimate Benben can now play back about half of the VGM files out there already.
+
+I got into the project mostly out of a love of old video game music, and partially because I was never totally happy with the official command line VGM player. I felt I could write a better player, but I also didn't want to write plain C or C++ code. Crystal felt like a perfect match, given its speed and capabilities. I could have just written Crystal bindings to the official VGM libraries, but that just didn't sound nearly as fun as writing a 100% native Crystal port.
+
+**Given your current experience with Crystal for kinda low-level stuff, what are the ups and the downs of writing such Crystal programs?**
+
+Oh I **love** Crystal for these sort of things. The type system in Crystal, especially type unions, gives it a very comfy feel, almost as if it was a dynamic language. I can write quick experimental code without worrying about types, and yet still get good performance out of it. Later on, I can take my initial code and tighten it up with type annotations. Plus, it gives me a lot of high-level tools within its standard library, but never prevents me from dropping down to manipulating pointers for those rare cases where it's needed. Essentially, I find it to be a perfect balance between high and low level programming languages.
+
+The only downside I've encountered so far is some of the error messages are sometimes a bit obscure. But those have gotten much better since the 0.3x days, and I'm sure they'll improve with time.
+
+**Completely unrelated, and on a personal note: why Remilia Scarlet?**
+
+Oh, the name? I am a huge fan of a game series called Touhou Project. One of my favorite characters from it is a vampire lady named Remilia. I started going by that online years ago, and got so used to it that I've just adopted it as my actual name. I guess I kind of identified with her lol. My actual real name is Alexa. Going by Remilia also means someone calling out to me won't trigger their Amazon Alexa device.
+
+**Back to your projects, do you know if there are people using them?**
+
+I'm not aware of anyone using my more recent audio-related projects. I'm hoping that will change, given that the libraries behind them could be a good option for music playback in video games.
+
+**You mean, for video games made in Crystal?**
+
+Yes, exactly! I've made a few prototype engines in Crystal, as well as an unfinished port of Doom, and feel that it could be a great language for writing video games based on my experiences with those.
+
+**About your Doom port! I remember you did some custom-made levels there, right?**
```

Comment:
This got changed but now it's weird. Changing it back (but I take other suggestions)

```suggestion
**I remember your Doom port! You did some custom-made levels there, right?**
```